pulp_rpm/app/viewsets.py

```diff
@@ -123,6 +123,70 @@ class RpmRepositoryViewSet(RepositoryViewSet, ModifyRepositoryActionMixin):
     endpoint_name = "rpm"
     queryset = RpmRepository.objects.exclude(user_hidden=True)
     serializer_class = RpmRepositorySerializer
+    queryset_filtering_required_permission = "rpm.view_rpmrepository"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list"],
+                "principal": ["authenticated"],
+                "effect": "allow",
+                "condition": "has_model_perms:rpm.view_rpmrepository",
+            },
+            {
+                "action": ["retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.view_rpmrepository",
+            },
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_perms:rpm.add_rpmrepository",
+            },
+            {
+                "action": ["update", "partial_update"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.change_rpmrepository",
+            },
+            {
+                "action": ["modify"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.change_rpmrepository",
+            },
+            {
+                "action": ["destroy"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.delete_rpmrepository",
+            },
+            {
+                "action": ["sync"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.sync_rpmrepository",
+            },
+        ],
+        "creation_hooks": [
+            {
+                "function": "add_roles_for_object_creator",
+                "parameters": {"roles": "rpm.rpmrepository_owner"},
+            }
+        ],
+    }
+
+    LOCKED_ROLES = {
+        "rpm.rpmrepository_owner": [
+            "rpm.add_rpmrepository",
+            "rpm.change_rpmrepository",
+            "rpm.delete_rpmrepository",
+            "rpm.view_rpmrepository",
+            "rpm.sync_rpmrepository",
+        ],
+    }
```

Comment:
Usually we have one 'creator', one 'owner' and one 'viewer' role. With the creator role targeted for model use, while the others are assigned by object.